contrib/example-backend/controllers/clusterbinding/clusterbinding_controller.go

```diff
@@ -19,300 +19,237 @@ package clusterbinding
-	config *rest.Config,
-	scope kubebindv1alpha2.InformerScope,
-	clusterBindingInformer bindinformers.ClusterBindingInformer,
-	serviceExportInformer bindinformers.APIServiceExportInformer,
-	clusterRoleInformer rbacinformers.ClusterRoleInformer,
-	clusterRoleBindingInformer rbacinformers.ClusterRoleBindingInformer,
-	roleBindingInformer rbacinformers.RoleBindingInformer,
-	namespaceInformer kubeinformers.NamespaceInformer,
-) (*Controller, error) {
-	queue := workqueue.NewTypedRateLimitingQueueWithConfig(workqueue.DefaultTypedControllerRateLimiter[string](), workqueue.TypedRateLimitingQueueConfig[string]{Name: controllerName})
+// ClusterBindingReconciler reconciles a ClusterBinding object.
+type ClusterBindingReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
 
-	logger := klog.Background().WithValues("controller", controllerName)
+	scope      kubebindv1alpha2.InformerScope
+	bindClient bindclient.Interface
+	reconciler reconciler
+}
 
+// NewClusterBindingReconciler returns a new ClusterBindingReconciler to reconcile ClusterBindings.
+func NewClusterBindingReconciler(
+	client client.Client,
+	scheme *runtime.Scheme,
+	config *rest.Config,
+	scope kubebindv1alpha2.InformerScope,
+	cache cache.Cache,
+) (*ClusterBindingReconciler, error) {
 	config = rest.CopyConfig(config)
 	config = rest.AddUserAgent(config, controllerName)
 
 	bindClient, err := bindclient.NewForConfig(config)
 	if err != nil {
 		return nil, err
 	}
-	kubeClient, err := kubeclient.NewForConfig(config)
-	if err != nil {
-		return nil, err
-	}
-
-	c := &Controller{
-		queue: queue,
-
-		clusterBindingLister:  clusterBindingInformer.Lister(),
-		clusterBindingIndexer: clusterBindingInformer.Informer().GetIndexer(),
-
-		serviceExportLister:  serviceExportInformer.Lister(),
-		serviceExportIndexer: serviceExportInformer.Informer().GetIndexer(),
-
-		clusterRoleLister:  clusterRoleInformer.Lister(),
-		clusterRoleIndexer: clusterRoleInformer.Informer().GetIndexer(),
-
-		clusterRoleBindingLister:  clusterRoleBindingInformer.Lister(),
-		clusterRoleBindingIndexer: clusterRoleBindingInformer.Informer().GetIndexer(),
-
-		namespaceLister:  namespaceInformer.Lister(),
-		namespaceIndexer: namespaceInformer.Informer().GetIndexer(),
 
+	r := &ClusterBindingReconciler{
+		Client:     client,
+		Scheme:     scheme,
+		scope:      scope,
+		bindClient: bindClient,
 		reconciler: reconciler{
 			scope: scope,
-			listServiceExports: func(ns string) ([]*kubebindv1alpha2.APIServiceExport, error) {
-				return serviceExportInformer.Lister().APIServiceExports(ns).List(labels.Everything())
+			listServiceExports: func(ctx context.Context, ns string) ([]*kubebindv1alpha2.APIServiceExport, error) {
+				var list kubebindv1alpha2.APIServiceExportList
+				if err := cache.List(ctx, &list); err != nil {
+					return nil, err
+				}
+				var exports []*kubebindv1alpha2.APIServiceExport
+				for i := range list.Items {
+					if list.Items[i].Namespace == ns || ns == "" {
```

Comment:
Why do we support clusterscoped and namespaced mode for the same resource? The original code didn't do it, did it?